src/Microsoft.Identity.Abstractions/DownstreamRestApi/AuthorizationHeaderProviderOptions.cs

```diff
@@ -0,0 +1,140 @@
+        /// </summary>
+        public string RelativePath { get; set; } = string.Empty;
+
+        /// <summary>
+        /// HTTP method used to call this downstream web API (by default Get).
+        /// </summary>
+        [DefaultValue("Get")]
+        public HttpMethod HttpMethod
+        {
+            get
+            {
+                return _httpMethod;
+            }
+            set
+            {
+                _httpMethod = value ?? throw new ArgumentNullException(nameof(value));
+            }
+        }
+
+        /// <summary>
+        /// Provides an opportunity for the caller app to customize the HttpRequestMessage. For example,
+        /// to customize the headers. This is called after the message was formed, including
+        /// the Authorization header, and just before the message is sent.
+        /// </summary>
+        public Action<HttpRequestMessage>? CustomizeHttpRequestMessage { get; set; }
+
+        /// <summary>
+        /// Options related to token acquisition.
+        /// </summary>
+        public AcquireTokenOptions AcquireTokenOptions
+        {
+            get
+            {
+                return _acquireTokenOptions;
+            }
+            set
+            {
+                _acquireTokenOptions = value ?? throw new ArgumentNullException(nameof(value));
+            }
+        }
+
+        /// <summary>
+        /// Name of the protocol scheme used to create the authorization header. By default: "Bearer".
+        /// </summary>
+        [DefaultValue("Bearer")]
+        public string ProtocolScheme
+        {
+            get
+            {
+                return _protocolScheme;
+            }
+            set
+            {
+                _protocolScheme = string.IsNullOrEmpty(value) ? throw new ArgumentNullException(_protocolScheme) : value;
+            }
+        }
+
+        /// <summary>
+        /// Describes if the downstream API is called on behalf of the calling service itself
+        /// (App token) or on behalf of a user processed by the service (user token).
+        /// If <c>true</c>, the token is requested on behalf of the app. Otherwise, it on-behalf of the user.
+        /// </summary>
+        /// <remarks>This is especially usefull when the call to the downstream API is described
+        /// by configuration.</remarks>
+        public bool RequestAppToken { get;set;}
+
+        /// <summary>
+        /// Clone the options (to be able to override them).
+        /// </summary>
+        /// <returns>A clone of the options.</returns>
+        public virtual AuthorizationHeaderProviderOptions Clone()
+        {
+            return new AuthorizationHeaderProviderOptions(this);
+        }
+
+        /// <summary>
+        /// Return the downstream web API URL.
+        /// </summary>
+        /// <returns>URL of the downstream web API.</returns>
+#pragma warning disable CA1055 // Uri return values should not be strings
+        public string GetApiUrl()
```

Comment:
What is the purpose of splitting the url into BaseUrl and RelativePath?

Reply:
One usage our customer wanted is base URL for the controller, and the relative path is the action.